File: app/src/main/java/com/androidexamples/app/fragment/FoodFragment.java
package com.androidexamples.app.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.androidexamples.app.R;
import com.androidexamples.app.interfaces.OnDetachFragmentListener;
import com.androidexamples.app.util.Constants;

/**
 * Created by marcos on 29/12/2016.
 */

public class FoodFragment extends Fragment {

    // Views
    private TextView textViewFood;
    private ImageView imageViewFood;
    private LinearLayout linearLayoutFood;

    // Another objects
    private OnDetachFragmentListener onDetachFragmentListener;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater layoutInflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return layoutInflater.inflate(R.layout.fragment_food, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        if(getArguments() != null) {
            textViewFood = (TextView)getView().findViewById(R.id.textViewFood);
            imageViewFood = (ImageView)getView().findViewById(R.id.imageViewFood);
            linearLayoutFood = (LinearLayout)getView().findViewById(R.id.linearLayoutFood);
            int resource = getArguments().getInt(Constants.KEY_RESOURCE);
            String text = getArguments().getString(Constants.KEY_TEXT);
            int color = getArguments().getInt(Constants.KEY_COLOR);
            setValues(resource, text, color);
        }

        setRetainInstance(true);
    }

    private void setValues(int resource, String text, int color) {
        linearLayoutFood.setBackgroundColor(ContextCompat.getColor(getActivity(), color));
        imageViewFood.setImageResource(resource);
        textViewFood.setText(text);
    }

    public void setOnDetachFragmentListener(OnDetachFragmentListener onDetachFragmentListener) {
        this.onDetachFragmentListener = onDetachFragmentListener;
    }

    @Override
    public void onDetach() {
        super.onDetach();

        if(onDetachFragmentListener != null)
            onDetachFragmentListener.onDetachFragment(getTag());
    }
}

File: app/src/main/java/com/androidexamples/app/flows/fragments/FoodFragment.kt
package com.androidexamples.app.flows.fragments

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.androidexamples.app.R
import com.androidexamples.app.interfaces.OnDetachFragmentListener
import com.androidexamples.app.utils.Constants

/**
 * Created by marcos on 29/12/2016.
 */

class FoodFragment : Fragment() {

    // Views
    private var textViewFood: TextView? = null
    private var imageViewFood: ImageView? = null
    private var linearLayoutFood: LinearLayout? = null

    // Another objects
    private var onDetachFragmentListener: OnDetachFragmentListener? = null

    override fun onCreateView(layoutInflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return layoutInflater.inflate(R.layout.fragment_food, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        if (arguments != null) {
            textViewFood = view!!.findViewById<View>(R.id.textViewFood) as TextView
            imageViewFood = view!!.findViewById<View>(R.id.imageViewFood) as ImageView
            linearLayoutFood = view!!.findViewById<View>(R.id.linearLayoutFood) as LinearLayout
            val resource = arguments!!.getInt(Constants.KEY_RESOURCE)
            val text = arguments!!.getString(Constants.KEY_TEXT)
            val color = arguments!!.getInt(Constants.KEY_COLOR)
            setValues(resource, text, color)
        }

        retainInstance = true
    }

    private fun setValues(resource: Int, text: String?, color: Int) {
        linearLayoutFood!!.setBackgroundColor(ContextCompat.getColor(activity!!, color))
        imageViewFood!!.setImageResource(resource)
        textViewFood!!.text = text
    }

    fun setOnDetachFragmentListener(onDetachFragmentListener: OnDetachFragmentListener) {
        this.onDetachFragmentListener = onDetachFragmentListener
    }

    override fun onDetach() {
        super.onDetach()

        if (onDetachFragmentListener != null)
            onDetachFragmentListener!!.onDetachFragment(tag!!)
    }
}
